doc/admin/troubleshooting.md

```diff
@@ -233,20 +233,62 @@ Refer to the DAOS Environment Variables document for
 more information about the debug system environment.
 
 ## Common DAOS Problems
-
+### Incompatible Agent ####
 When DER_AGENT_INCOMPAT is received, it means that the client library libdaos.so
 is likely mismatched with the DAOS Agent.  The libdaos.so, DAOS Agent and DAOS
 Server must be built from compatible sources so that the GetAttachInfo protocol
 is the same between each component.  Depending on your situation, you will need
 to either update the DAOS Agent or the libdaos.so to the newer version in order
 to maintain compatibility with each other.
 
+### HLC Sync ###
 When DER_HLC_SYNC is received, it means that sender and receiver HLC timestamps
 are off by more than maximum allowed system clock offset (1 second by default).
 
 In order to correct this situation synchronize all server clocks to the same
 reference time, using services like NTP.
 
+### Shared Memory Errors ###
+When DER_SHMEM_PERMS is received it means that this IO Engine lacked the
+permissions to access the shared memory megment left behind by a previous run of
+the IO Engine on the same machine.  This happens when the IO Engine fails to
+remove the shared memory segment upon shutdown, and, there is a mismatch between
+the user/group used to launch the IO Engine between these successive runs.  To
+remedy the problem, manually identify the shared memory segment and remove it.
+
+Issue ```ipcs``` to view the Shared Memory Segments.  The output will show a
+list of segments organized by ```key```.
+
+```
+$ipcs
+
+------ Message Queues --------
+key        msqid      owner      perms      used-bytes   messages
+
+------ Shared Memory Segments --------
+key        shmid      owner      perms      bytes      nattch     status
+0xffffffff 49938432   root       666        40         0
+0x10242048 98598913   jbrosenz   660        1048576    0
+0x10242049 98631682   jbrosenz   660        1048576    0
+
+------ Semaphore Arrays --------
+key        semid      owner      perms      nsems
+```
+
+Shared Memory Segments with keys [0x10242048 .. (0x10242048 + number of IO
+Engines running)] are the segments that must be removed.  Use ```ipcrm``` to
+remove the segment.
+
+For example, to remove the shared memory segment left behind by IO Engine
+instance 0, issue:
+```
+sudo ipcrm -M 0x10242048
+```
+To remove the shared memory segment left behind by IO Engine instance 1, issue:
+```
+sudo ipcrm -M 0x10242049
+```
+
```

Comment:
Any need to leave in discussion of the other case, where the shared memory is full?

Reply:
My thought was that it was not necessary because it really ought to be caught by developers.  I had it in originally for completeness, but I think that now there is a specific error code for the permissions failure, it is cleaner this way.

Reply:
Okay, works for me. Thanks.